### PR TITLE
Deploy native token from lottery.

### PR DESCRIPTION
**Pull request by rand0c0des**

### CODE REVIEW

File: src/Lottery.sol [hunk 1]
```diff
@@ -39,6 +41,10 @@ contract Lottery is ILottery, Ticket, LotterySetup, ReferralSystem, RNSourceCont
     mapping(uint128 => uint256) public override ticketsSold;
     int256 public override currentNetProfit;
 
+    ILotteryToken public immutable nativeToken;
```
Comment (rand0c0des) [inline src/Lottery.sol:44]: Add this to interface

File: src/Lottery.sol [hunk 2]
```diff
@@ -39,6 +41,10 @@ contract Lottery is ILottery, Ticket, LotterySetup, ReferralSystem, RNSourceCont
...
+
+    uint256[] private inflationRatesPerDraw;
```
Comment (rand0c0des) [inline src/Lottery.sol:46]: Make it public and add to interface?

File: src/Lottery.sol [hunk 3]
```diff
@@ -74,33 +80,38 @@ contract Lottery is ILottery, Ticket, LotterySetup, ReferralSystem, RNSourceCont
     }
 
     /// @dev Constructs a new lottery contract.
-    /// @param lotterySetupParams Setup parameter for the lottery
-    /// @param lotteryToken Lottery native token address.
+    /// @param lotterySetupParams Setup parameter for the lottery.
     /// @param percentageRewardsToPlayers Percentage of native token rewards going to players.
+    /// @param _inflationRatesPerDraw The inflation rates of native token per draw for each year.
```
Comment (rand0c0des) [inline src/Lottery.sol:85]: Order of cmments to match order of args

**Review by 0xluckydev — DISMISSED** [review]

File: src/Lottery.sol (hunk 2 above)
Comment (0xluckydev) [reply src/Lottery.sol:46]: Not sure if we need it public

**Review by 0xluckydev — APPROVED** [review]

**Pull request merged**